duckduckgo/Android
# Feature/downloading data uri

Task/Issue URL: https://app.asana.com/0/547792610048271/565680240942081
Tech Design URL: 
CC: 

**Description**:
Adds support for saving images encoded using a data URI.

**Steps to test this PR**:
1. Visit a site which renders an image as a data URI (e.g., https://dopiaza.org/tools/datauri/examples/index.php)
1. Long press on the image, and ensure it can be saved
1. During the decoding, a notification will show that the image is downloading - similar to downloading normal images.
1. Once decoded and written to disk, the notification will show that the image is downloaded.

⚠️  Ensure you test on an Oreo device and a pre-Oreo device; notifications changed then. If the notification isn't registered to an appropriate channel upfront, the notification will never show.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
